### ISSUE TITLE
Typo in Shopping Cart lesson url in production, but not locally

### ISSUE THREAD

**Issue by rankoliang** [TheOdinProject/theodinproject#1669]
The url of the shopping cart assignment is currently https://www.theodinproject.com/courses/javascript/lessons/shopping-chart when it should be
https://www.theodinproject.com/courses/javascript/lessons/shopping-cart. (chart instead of cart)

The routes seem to be working correctly on a local server, however.
![image](https://user-images.githubusercontent.com/37024941/105614985-121aea80-5d9b-11eb-9c7c-d593cf0f6033.png)

In addition, the file in the curriculum and its reference in the seeds file also have this spelling mistake.
https://github.com/TheOdinProject/theodinproject/blob/35dd141b0f9c6ba1a38e145373a744ccaaaa71fd/db/seeds/06_javascript_course_seeds.rb#L468

https://github.com/TheOdinProject/curriculum/blob/master/javascript/react-js/project-shopping-chart.md

**Comment by marvingay** [comment]
The misspelling was caught after this course's release and it was left in place as it would have been a bit of an inconvenience to switch in the advent of the project submissions feature, I believe.

The spelling error is trivial, really, but perhaps @KevinMulhern  can shed a more accurate light on this.

**Comment by rankoliang** [comment]
I believe I've figure out the issue. It's that the Lesson got assigned the original id of `Shopping Chart` when it got created. When the title was manually updated in the seed via the create_or_update_lesson method below, it kept the same incorrectly spelt id and builds the incorrectly spelt path names. https://github.com/TheOdinProject/theodinproject/blob/146065f9fbbe78753be663da98d41f4a08b9c36e/db/seeds.rb#L61

**Comment by KevinMulhern** [comment]
Nice catch @rankoliang 💪  thanks for reporting this.

The route path is derived from the title attribute of the lesson. Friendly id uses it to create a slug when the record is first created. I assumed it would update the slug automatically when we updated the title to correct the spelling mistake. But that doesn't seem to be working as expected. 

I think that behaviour has changed in the newer version of friendly_id and we need to add a `should_generate_new_friendly_id ` method to the lesson model to get that functionality back.
https://norman.github.io/friendly_id/file.Guide.html#Deciding_When_to_Generate_New_Slugs

For now,  I've manually updated the slug for the shopping cart lesson in production to resolve this particular issue. Thanks again for reporting @rankoliang, it's very much appreciated.
